core/dbt/events/functions.py

```diff
@@ -133,15 +133,16 @@ def event_to_serializable_dict(
     node_info = dict()
     if hasattr(e, '__dataclass_fields__'):
         for field, value in dataclasses.asdict(e).items():  # type: ignore[attr-defined]
-            _json_value = e.fields_to_json(value)
+            if field not in ["code", "report_node_data"]:
+                _json_value = e.fields_to_json(value)
 
-            if isinstance(e, NodeInfo):
-                node_info = dataclasses.asdict(e.get_node_info())
+                if isinstance(e, NodeInfo):
+                    node_info = dataclasses.asdict(e.get_node_info())
```

Comment:
I know I wrote this - but is there a reason it can't be above the `for` on 135 so we don't hit it possibly multiple times?

Reply:
haha yeah let's fix that here.